homestar-runtime/tests/utils.rs

```diff
@@ -85,9 +88,31 @@ pub(crate) fn stop_all_bins() -> Result<()> {
     Ok(())
 }
 
+/// Retrieve process output.
+pub(crate) fn retrieve_output(proc: Child) -> String {
+    let output = proc.wait_with_output().expect("failed to wait on child");
+    let plain_stdout_bytes = strip_ansi_escapes::strip(output.stdout);
+    String::from_utf8(plain_stdout_bytes).unwrap()
+}
+
+/// Kill the Homestar server/binary
+pub(crate) fn kill_homestar(mut homestar_proc: Child) -> Child {
+    if let Ok(None) = homestar_proc.try_wait() {
```

Comment:
without the extra stop, I wonder if there's an else or other cases to check here?

Reply:
My understanding is that if `try_wait` returns `Ok(Some(status))`, then the child has already exited so we should be safe without an else condition. 

In cases where we don't call `stop_homestar`, we will enter this condition every time and attempt to kill the process after a one second timeout. We probably don't want the timeout in this case. I think it was originally intended for cases where you call `stop_homestar` first and want to wait for that to resolve.

We could call `stop_homestar` every time, but I'm not certain what will happen in cases where we have multiple nodes running. Will `stop_homestar` stop them all? Alternatively, we could parameterize the wait duration to `kill_homestar` and kill right away where it makes sense.

Reply:
@bgins yep. this was a bit before we found the issue that you fixed :).

Reply:
Added a timeout parameter to `kill_homestar` to give us flexibility on how long we wait.